packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -198,14 +198,25 @@ class DependencyExtractionWebpackPlugin {
 				}
 			}
 
+			// Go through the assets and hash the sources. We can't just use
+			// `entrypointChunk.contentHash` because that's not updated when
+			// assets are minified. Sigh.
+			// @todo Use `asset.info.contenthash` if we can make sure it's reliably set.
+			const hash = createHash( 'sha512' );
+			for ( const filename of entrypoint.getFiles().sort() ) {
+				const asset = compilation.getAsset( filename );
+				hash.update( `${ filename }: ` );
+				asset.source.updateHash( hash );
+			}
+
 			const entrypointChunk = isWebpack4
```

Comment:
We can now move `entrypointChunk` constant further in the code and closer to the usage.

Reply:
Actually, we probably could get files from the `entrypointChunk.files` rather than from `entrypoint.getFiles()` to align with the handling in other places:

```js
const entrypointChunk = isWebpack4
	? entrypoint.chunks.find( ( c ) => c.name === entrypointName )
	: entrypoint.getEntrypointChunk();

const entrypointChunkHash = createHash( 'sha512' );
for ( const filename of Array.from( entrypointChunk.files ).sort() ) {
	entrypointChunkHash.update( filename + ':' + compilation.getAsset( filename ).source.source() );
}
```

Reply:
In `@wordpress/scripts` there is this logic:

https://github.com/WordPress/gutenberg/blob/67fcf07c2da488f1e5ff9bc0dda648388911b35c/packages/scripts/config/webpack.config.js#L107-L123

With the current implementation it processes 3 files:

```
[ './style-index.css', 'index.css', 'index.js' ]
```

`Array.from( entrypointChunk.files )` processes only 2 files because `./style-index.css` goes to its own chunk:

```
[ 'index.css', 'index.js' ]
```

Reply:
Are you still suggesting to change the implementation here as in your second comment, or is the third comment convincing yourself not to?

It seems to me that your last comment is the argument not to make the change you suggested in your second comment. The extra file is part of the asset and so should be covered by the hash.

Reply:
My last comment provides the reasoning why the change included in https://github.com/WordPress/gutenberg/pull/34969#discussion_r865656919 would be a good improvement. We don't care about `./style-index.css` that goes into its chunk so it shouldn't matter when calculating the hash for the JS entry point.

Reply:
Would you care if I point out that someone using `.optimization.runtimeChunk` would find that the `runtime.js` or `runtime~index.js` isn't included in the version hash either?

If you really want this change I'm not going to fight it since our use case has neither of these sorts of extra files. But I do think if we're going to have one hash for the asset, it should cover the whole asset rather than excluding parts of it.

Reply:
There's also the possibility that someone is using splitChunks like you do there but on JS code chunks, or with Webpack's automatic vendor splitting.